### PR TITLE
Update default k8s version to v1.25 for testing

### LINKED ISSUE #3055: Remove k8s 1.22 tests and add 1.26
Since k8s 1.26 is now out and 1.22 EOL, we should:

- [x]  remove https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capi-periodic-upgrade-main-1-22-1-23
- [x] add a new 1.25 to 1.26 k8s upgrade periodic job
- [x] remove the CSI migration test spec and associated presubmit as it's only valid for 1.22 to 1.23 upgrade
- [x] bump the default k8s versions used in testing (e2e, testgrid, etc.)

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

**What this PR does / why we need it**:

This PR updates the version of Kubernetes we test by default to v1.25 from v1.24. Correspondingly, we also will upgrade to v1.26 instead of v1.25.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #3055

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -153,13 +153,13 @@ providers:
 variables:
   AKS_KUBERNETES_VERSION: "latest"
   AKS_KUBERNETES_VERSION_UPGRADE_FROM: "latest-1"
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-stable-1.24}"
-  FLATCAR_KUBERNETES_VERSION: "${FLATCAR_KUBERNETES_VERSION:-stable-1.24}"
+  KUBERNETES_VERSION: "${KUBERNETES_VERSION:-stable-1.25}"
+  FLATCAR_KUBERNETES_VERSION: "${FLATCAR_KUBERNETES_VERSION:-stable-1.25}"
```
Comment (jackfrancis) [inline test/e2e/config/azure-dev.yaml:157]: @invidian is there any reason not to update the default k8s version to test our flatcar reference cluster scenario against?

Comment (invidian) [reply test/e2e/config/azure-dev.yaml:157]: Should be fine to update. I'll be making sure that at all new minor versions are available. It seems version discovery works as expected, as 1.25.4 was already available. I'll build 1.25.6 with new stable release now.

**Comment by jackfrancis** [comment]
/retest

Another GPU flake

@mboersma is it worth auditing recent GPU failures to sanity check our regional configuration? Maybe GPU availability has diverged from our prior understanding of regional availability for the SKU we're testing?

**Comment by jackfrancis** [comment]
O.K., in response to these GPU flakes:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3099

**Comment by jackfrancis** [comment]
Maybe k8s v1.25 doesn't work with this version of gpu-operator? Let's merge https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3099 first (which is passing w/ v1.24) and see if this change rebased on top of it works w/ 1.25.

**Comment by jackfrancis** [comment]
/test pull-cluster-api-provider-azure-e2e-workload-upgrade
/test pull-cluster-api-provider-azure-e2e-optional

This seems to now pass w/ the newer gpu-operator.

cc @CecileRobertMichon @mboersma for a review round

**Review by mboersma — COMMENTED** [review]
/lgtm
/assign @CecileRobertMichon

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**